woodwork/utils.py

```diff
@@ -298,3 +299,149 @@ def _parse_logical_type(logical_type, name):
+
+            # Combine table names
+            if obj.ww.name is not None:
+                if table_name:
+                    table_name += '_'
+                table_name += str(obj.ww.name)
+
+            # Handle table indexes
+            for idx_type, obj_idx, table_idx in [('index', obj.ww.index, index),
+                                                 ('time_index', obj.ww.time_index, time_index)]:
+                if obj_idx is not None:
+                    if table_idx is None:
+                        if idx_type == 'index':
+                            index = obj_idx
+                        elif idx_type == 'time_index':
+                            time_index = obj_idx
+                    elif obj_idx != table_idx:
+                        raise IndexError(f'Cannot concat dataframes with different Woodwork {idx_type} columns.')
+                    elif axis in {1, 'columns'} and obj_idx == table_idx:
+                        # Do not include the column from the dataframe to avoid duplicates columns
+                        drop_cols.append(table_idx)
+
+            # To avoid changing input objects, create a new DataFrame without the index columns
+            if drop_cols:
+                obj_to_add = obj.ww.drop(columns=drop_cols)
+            ww_columns = obj_to_add.ww.schema.columns
+        elif isinstance(obj.ww.schema, ww.column_schema.ColumnSchema):
+            ww_columns = {obj.name: obj.ww.schema}
+
+        concat_objs.append(obj_to_add)
+
+        # Compile the typing information per column
+        for name, col_schema in ww_columns.items():
+            # Do not look at typing information for duplicate columns.
+            # Means that index columns will only have first occurrance's typing information
+            if name in col_names_seen:
+                continue
+            logical_types[name] = col_schema.logical_type
+            semantic_tags[name] = col_schema.semantic_tags - {'time_index'} - {'index'}
+            col_metadata[name] = col_schema.metadata
+            col_descriptions[name] = col_schema.description
+            use_standard_tags[name] = col_schema.use_standard_tags
+
+            col_names_seen.add(name)
+
+    # Perform concatenation with the correct library
+    dd = import_or_none('dask.dataframe')
+    ks = import_or_none('databricks.koalas')
+
+    lib = pd
+    if ks and isinstance(obj, (ks.Series, ks.DataFrame)):
+        lib = ks
+    elif dd and isinstance(obj, (dd.Series, dd.DataFrame)):
+        lib = dd
+
+    combined_df = lib.concat(concat_objs, axis=axis, join=join)
+
+    # Cannot initialize Woodwork on series output
+    # --> we can handle if we really want to we'll just need to account for this case when collecting typging info
+    if not isinstance(combined_df, lib.DataFrame):
+        warnings.warn('Woodwork not initialized on the output of ww.concat', ww.exceptions.WoodworkNotInitWarning)
+        return combined_df
```

Comment:
The situation where a user concatenates multiple series along `axis=0` is the only way we get a series as output from concat, and we can initialize Woodwork by choosing a similar approach where we only use the first column's typing information, but it would add enough further complexity here, that I figured it might make sense to wait until we have a solid use case for it.

Reply:
Made a related comment about this in a test. Not sure if this would be a common use case or not. In any case, if we leave as is, I would just make sure the associated comments indicate that we are not supporting this currently, not so much that we cannot support it.

Reply:
Situation no longer possible now that we restrict `axis=1`